File: problem2.py
class Solution:
    def isIsomorphic(self, s: str, t: str) -> bool:
        if len(s) != len(t) or len(s) == 0 or s == None:
            return False
        
        sMap = {}
        tMap = {}

        for i in range(len(s)):
            sChar = s[i]
            tChar = t[i]

            if sChar in sMap:
                if sMap.get(sChar) != tChar:
                    return False
            else:
                sMap[sChar] = tChar
            if tChar in tMap:
                if tMap.get(tChar) != sChar:
                    return False
            else:
                tMap[tChar] = sChar
        return True

File: problem2.java
// The solution utilizes HashMap data-structure to map characters from string s to string t
class Solution {
    public boolean isIsomorphic(String s, String t) {
        if(s.length() != t.length() || s == null || s.length() == 0) {
            return false;
        }

        HashMap<Character, Character> sMap = new HashMap<>();
        //HashMap<Character, Character> tMap = new HashMap<>();
        HashSet<Character> set = new HashSet<>();
        for(int i=0; i< s.length(); i++) {
            char sChar = s.charAt(i);
            char tChar = t.charAt(i);
            if(sMap.containsKey(sChar)) {
                if(sMap.get(sChar) != tChar) {
                    return false;
                }
            }
            else {
                sMap.put(sChar, tChar);
                if(set.contains(tChar)) {
                    return false;
                }
                set.add(tChar);
            }
        }
        return true;
    }
}